meson.build

```diff
@@ -175,7 +175,7 @@ if get_option('fuzzing')
     endif
 endif
 
-incdirs = include_directories('include', 'compat')
+incdirs = include_directories('.', 'include', 'compat')
```

Comment:
Meson by default adds includes for the current build dir and source dir, so I'm curious why this was needed (was it?). Shouldn't hurt either of course.

Reply:
Without this the `#include "config.h"` were failing:
```
[2/14] cc -Itests/59830eb@@unittest@exe -Itests -I../tests -I../include -I../compat -fdiagnostics-color=always -DNDEBUG -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -O3 -Wsign-compare -Wstrict-aliasing -Wstrict-overflow -Wformat=2 -Wno-unused-parameter -Wcast-align -Wpointer-arith -Wmissing-prototypes -Wwrite-strings -Wlogical-op -DHAVE_CONFIG_H -MD -MQ 'tests/59830eb@@unittest@exe/unittests.c.o' -MF 'tests/59830eb@@unittest@exe/unittests.c.o.d' -o 'tests/59830eb@@unittest@exe/unittests.c.o' -c ../tests/unittests.c
```
This is missing a `-I../` (although I'd prefer it to be an absolute path...)

Reply:
Oh right, I missed that this was at the top-level and you added subdir `meson.build`. Makes perfect sense then.

Reply:
This looks more like a missing dependency, you should register this in a dependency and add it:

```
configure_file(output : 'config.h', configuration : cdata)
````

Check https://github.com/videolabs/libdsm/blob/master/meson.build#L146 ;)

Reply:
This is already done in the main meson.build, and the config.h is properly generated in the build folder, unless you can the returned object to the list of dependencies and meson can figure out that it requires an additional `-I` parameter?

Reply:
That's not really required in this case because `configure_file` will be output at configure time, so it's always guaranteed to be there when the build starts. Only needed if you create headers by other means like custom targets or such.

Reply:
Actually, it seems the documentation is adding the `.` anyway : https://mesonbuild.com/Configuration.html#a-full-example